chatbot2.js

```diff
@@ -495,78 +495,80 @@ client.on("chat", (channel, userstate, message, self) => {
   //no msg-id when normal msg
   //msg-id: "highlighted-message"
   //msg-id: "skip-subs-mode-message"
-  if (cmd == "!langs") {
-    let languages = Object.keys(ttsLangs);
-    let langlist = "";
-    for (var i = 0; i < languages.length; i++) {
-      langlist = langlist + " " + languages[i];
-    }
-    client.say(config.credentials.channelName, langlist);
-  }
-  if (cmd == "!ignore") {
-    if (
-      userstate["mod"] ||
-      userstate["username"] == config.credentials.channelName
-    ) {
+  switch(cmd) {
```

Comment:
Instead of switch i was thinking about creating object with commands and then just loop over commands in object and find which one matches "cmd" and execute function associated with it. Also i will move most functions from chatbot2 to modules to keep code cleaner.

Reply:
It'd definitely work if you did it that way but looping over commands is unnecessary when you can just use ifs or a switch, moving all the command logic to a separate file and just calling `processCmd(cmd)` or something in the main .js file would be a big readability improvement though

Reply:
Doing some reading on [high-performance JS](https://www.oreilly.com/library/view/high-performance-javascript/9781449382308/ch04.html#if-else_versus_switch)
Switch is faster than if-else but not as fast as array lookup, but I don't think theres a good way to do array lookups for the cmd logic